run.c

```diff
@@ -82,86 +138,50 @@ void run(void)
          * No need to do a domain decomposition as we already did one since
          * the last move in compute_accelerations().
          * This is after advance_and_find_timesteps so the acceleration
          * is included in the kick.*/
-        if(All.Ti_Current >= All.Ti_nextoutput)
+        if(WillOutput)
         {
             /*Save snapshot*/
             savepositions(All.SnapshotFileCount++, action == NO_ACTION);	/* write snapshot file */
-            All.Ti_nextoutput = find_next_outputtime(All.Ti_nextoutput + 1);
-        }
-        write_cpu_log();		/* produce some CPU usage info */
-
-        if(action == STOP || action == TIMEOUT) {
-            /* OK snapshot file is written, lets quit */
-            return;
-        }
-
-        All.NumCurrentTiStep++;
-
-        report_memory_usage("RUN");
-
-        action = human_interaction();
-        switch(action) {
-            case STOP:
-                message(0, "human controlled stop with checkpoint.\n");
-                All.Ti_nextoutput = All.Ti_Current;
-                /* Note there is an error involved in doing this:
-                 * part of the SPH VelPred array is computed using
-                 * the length of a PM step, and will have been
-                 * slightly incorrect for this timestep. But we want a stop NOW.*/
-                All.PM_Ti_endstep = All.Ti_Current;
-                /* next loop will write a new snapshot file; break is for switch */
-                break;
-            case TIMEOUT:
-                message(0, "stopping due to TimeLimitCPU.\n");
-                All.Ti_nextoutput = All.Ti_Current;
-                All.PM_Ti_endstep = All.Ti_Current;
-                /* next loop will write a new snapshot file */
-                break;
-
-            case AUTO_CHECKPOINT:
-                message(0, "auto checkpoint due to TimeBetSnapshot.\n");
-                All.Ti_nextoutput = All.PM_Ti_endstep;
-                /* will write a new snapshot file next time the PM step finishes*/
-                break;
-
-            case CHECKPOINT:
-                message(0, "human controlled checkpoint.\n");
-                All.Ti_nextoutput = All.PM_Ti_endstep;
-                /* will write a new snapshot file next time the PM step finishes*/
-                break;
 
-            case TERMINATE:
-                message(0, "human controlled termination.\n");
-                /* no snapshot, at termination, directly end the loop */
+            if(action == STOP || action == TIMEOUT) {
+                /* OK snapshot file is written, lets quit */
                 return;
+            }
 
-            case IOCTL:
-                break;
-
-            case NO_ACTION:
-                break;
+            /*Do the extra half-kick we avoided for a snapshot.*/
+            apply_half_kick();
+            /*Find next output*/
+            Ti_nextoutput = find_next_outputtime(Ti_nextoutput);
```

Comment:
Does this have to be after apply_half_kick? It is related to the previous comment. If it has to be then we can probably run it every step.